### PR TITLE
log rank zero everywhere

### PR DESCRIPTION
**Pull request by RdoubleA**
#### Context
Are you tired of constantly checking for rank zero before logging? Then `log_rank_zero` is just the solution for you!

#### Changelog
What are the changes made in this PR?
* Remove pesky if statements checking for rank zero before logging and instead use `log_rank_zero`
* add `stacklevel=2` to `log_rank_zero` so the line from the calling function is shown instead of within `log_rank_zero`. This is not yet relevant, until we decided to show more details on where the log was emitted by configuring the logger format more, but this can improve UX internally

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

File: torchtune/utils/_logging.py [hunk 1]
```diff
@@ -98,4 +98,4 @@ def log_rank_zero(logger: logging.Logger, msg: str, level: int = logging.INFO) -
     rank = dist.get_rank() if dist.is_available() and dist.is_initialized() else 0
     if rank != 0:
         return
-    logger.log(level, msg)
+    logger.log(level, msg, stacklevel=2)
```
Comment (joecummings) [inline torchtune/utils/_logging.py:101]: bueno

**Review by joecummings — APPROVED** [review]
🫡

File: recipes/full_finetune_distributed.py [hunk 2]
```diff
@@ -497,11 +497,10 @@ def _setup_model(
         # Ensure no params and buffers are on meta device
         training.validate_no_params_on_meta_device(model)
 
-        utils.log_rank_zero(
-            log,
-            f"Instantiating model and loading checkpoint took {time.perf_counter() - init_start:.2f} secs",
-        )
         if self._is_rank_zero:
+            log.info(
```
Comment (joecummings) [inline recipes/full_finetune_distributed.py:501]: Wait why can this not be extracted out?

Comment (RdoubleA) [reply recipes/full_finetune_distributed.py:501]: `init_start` is only defined for rank zero, and the string formatting happens before we check for rank zero in `log_rank_zero`
